### DISCUSSION TITLE
Bug: QueryParams is not working

### DISCUSSION THREAD

**Discussion by a0m0rajab** [supabase/supabase#20439] [Contribute to Supabase]
# Bug report

- [ ] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

In this PR: https://github.com/open-sauced/insights/pull/1184 
We are using [auto-signin](https://supabase.com/docs/reference/javascript/auth-signinwithoauth) from SupaBase with redirect and queryParams option, I am expecting the QueryParams to be parsed and passed with react but it's not. 

Not sure, if this is a bug or the QueryParams option is expected to do something different. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to '…'
2. Click on '…'
3. Scroll down to '…'

## Expected behavior

Eexpected to be able to get the query params in the redirected URL. 

## System information

- OS: [e.g. macOS, Windows]: windows
- Browser (if applies) [e.g. chrome, safari] chrome
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by a0m0rajab** [comment]
Related [discussion](https://github.com/orgs/supabase/discussions/9845):

**Comment by mdjohns** [comment]
is the expected behavior here that the `queryParams` object is appended to the `redirectTo` URL?

If so, it looks like [this is the culprit](https://github.com/supabase/gotrue-js/blob/0f04bfc30e9cdd841ec723448142d8e0dcb63709/src/GoTrueClient.ts#L1766-L1772). `queryParams` is appended to the end of the whole OAuth URL, not the `redirectTo` URL.

**Comment by a0m0rajab** [comment]
For me, yeah I was expecting it to be appended to the redirect to URL.

**Comment by mdjohns** [comment]
[The docs](https://supabase.com/docs/reference/javascript/auth-signinwithsso) are not super clear here:
> `queryParams` An object of query params

I think we need a member of the team to chime in on the expected behavior here.

If we are expecting `queryParams` to be appended to the whole URL (I'm assuming to be consumed by the OAuth provider), then we should probably specify that.

If we are expecting `queryParams` to be appended to the `redirectTo` URL, then I think we need to make some changes to the code linked above.

**Comment by a0m0rajab** [comment]
And it would be great to have some idea about appending the query Params in the redirectTo if it was to be consumed with the OAuth provider.

We solved this manually before, by appending a string to the redirect to URL

Sent from mobile device


________________________________

**Comment by leonardo2204** [comment]
Is this dead? It seems to be a bug to me...
